lib/src/puppetdb-cli.cc

```diff
@@ -107,7 +121,85 @@ PuppetDBConn::parseServerUrls(const json::JsonContainer& config) {
     }
 }
 
-size_t write_data(void *ptr, size_t size, size_t nmemb, FILE *stream) {
+class JsonQueue {
+  public:
+    char front() {
+        boost::unique_lock<boost::mutex> mlock(mutex_);
+        cond_.wait(mlock, [&]{ return !queue_.empty(); });
+        return queue_.front();
+    }
+
+    char pop() {
+        boost::unique_lock<boost::mutex> mlock(mutex_);
+        cond_.wait(mlock,  [&]{ return !queue_.empty(); });
+        auto item = queue_.front();
+        queue_.pop();
+        return item;
+    }
+
+    void push(char* items, size_t num_items) {
```

Comment:
We'd have to template the class definition in that case, I'd prefer we stick with a vector<char> until we need something more.